scripts/build/Jenkins/Jenkinsfile

```diff
@@ -336,51 +416,281 @@ def PreBuildCommonSteps(Map pipelineConfig, String snapshot, String repositoryNa
+            def the_branch = branchName
+            if(EXTRAS_COMMIT) {
+                //the pr is for the engine repo, so we want to use the passed in branch name and record the change id and change date
+                setEnvCommit = true
+            } else {
+                //this pr is for another repo so we want to use the development branch for the engine and do not set the commit id or change date
+                the_branch = EXTRAS_DEVELOPMENT_BRANCH
+            }
 
-        // Always run the clean step, the scripts detect what variables were set, but it also cleans if
-        // the NODE_LABEL has changed
-        def command = "${pipelineConfig.PYTHON_DIR}/python"
-        if(env.IS_UNIX) command += '.sh'
-        else command += '.cmd'
-        command += " -u ${pipelineConfig.BUILD_ENTRY_POINT} --platform ${platform} --type clean"
-        palSh(command, "Running ${platform} clean")
+            UnInstallGitLFS() // Prevent git from pulling lfs objects during checkout
+
+            //checkout the engine branch
+            CheckoutRepo(the_branch, GetExtrasRemoteConfig(), disableSubmodules, setEnvCommit)
+
+            InstallAndPullGitLFS()
+        }
+        //get any other external repos next, set the env.CHANGE_ID and env.CHANGE_DATE only if the repo is the pr repo
+        pipelineConfig.EXTERNAL_REPOS.each { repo ->
+            dir(repo.REPOSITORY_NAME) {
+                echo "Checkout External Repo: ${repo.URL}"
+                def setEnvCommit = false
+                def the_branch = branchName
+                if(!ENGINE_COMMIT && !EXTRAS_COMMIT) {
+                    //the pr is for the this repo, so we want to use the passed in branch name and record the change id and change date
+                    setEnvCommit = true
+                } else {
+                    //this pr is for the another repo so we want to use the development branch for the this repo and do not set the commit id or change date
+                    the_branch = repo.DEVELOPMENT_BRANCH
+                }
+
+                UnInstallGitLFS() // Prevent git from pulling lfs objects during checkout
 
-        if(fileExists('.lfsconfig')) {
-            palSh("git lfs install", "LFS config exists. Installing LFS hooks to local repo")
-            palSh("git lfs pull", "Pulling new LFS objects")
+                //checkout the branch
+                CheckoutRepo(the_branch, GetRemoteConfig(repo.REPOSITORY_NAME, repo.URL), disableSubmodules, setEnvCommit)
+
+                InstallAndPullGitLFS()
+            }
         }
     }
 }
 
 def Build(Map pipelineConfig, String platform, String type, String workspace) {
     timeout(time: env.TIMEOUT, unit: 'MINUTES', activity: true) {
         def command = "${pipelineConfig.PYTHON_DIR}/python"
-        if(env.IS_UNIX) command += '.sh'
+        def ext = ''
+        if(env.IS_UNIX) {
+            command += '.sh'
+            ext = '.sh'
+        }
         else command += '.cmd'
```

Comment:
In the Windows case, should we just make the `ext = .bat`?

Reply:
This calls the build scripts, which are all .cmd files: https://github.com/o3de/o3de/tree/development/scripts/build/Platform/Windows

Reply:
yeas, .cmd is correct